esti/presign_multipart_test.go

```diff
@@ -171,84 +174,172 @@ func TestCompletePresignMultipartUpload(t *testing.T) {
 			require.Equalf(t, tt.statusCode, resp.StatusCode(), "CompletePresignMultipartUpload status code mismatch (expected %d): %s - %s", tt.statusCode, resp.Status(), resp.Body)
 		})
 	}
+}
 
-	t.Run("valid", func(t *testing.T) {
-		// not a short test
-		if testing.Short() {
-			t.Skip()
-		}
-		const (
-			objPath       = "presign_multipart_upload/complete"
-			numberOfParts = 2
-		)
-		respCreate, err := client.CreatePresignMultipartUploadWithResponse(ctx, repo, mainBranch, &apigen.CreatePresignMultipartUploadParams{
-			Path:  objPath,
-			Parts: swag.Int(numberOfParts),
-		})
-		require.NoError(t, err)
-		require.NotNil(t, respCreate.JSON201)
+func TestPresignMultipartUploadSeparateParts(t *testing.T) {
+	skipPresignMultipart(t)
+	// not a short test
+	if testing.Short() {
+		t.Skip()
+	}
 
-		// upload parts
-		httpClient := http.Client{
-			Timeout: 30 * time.Second, // make sure we do not wait forever
-		}
-		var parts []apigen.UploadPart
-		for i := 0; i < numberOfParts; i++ {
-			startTime := time.Now()
-			// random data - all parts except the last one should be at least >= MinUploadPartSize
-			n, err := rand.Int(rand.Reader, big.NewInt(1<<20))
-			require.NoError(t, err)
-			var partSize int64
-			if i < numberOfParts-1 {
-				partSize = manager.MinUploadPartSize + n.Int64() // 5mb + ~1mb
-			} else {
-				partSize = n.Int64() + 1 // ~1mb + 1
+	ctx, _, repo := setupTest(t)
+	defer tearDownTest(repo)
+
+	// Create a large object.
+	r := mathrand.New(mathrand.NewSource(17))
+	data := make([]byte, largeDataContentLength)
+	_, err := r.Read(data)
+	require.NoError(t, err)
+	const largeObjectPath = "data/large"
+	_, err = helpers.ClientUpload(ctx, client, repo, mainBranch, largeObjectPath, nil, "", bytes.NewReader(data))
+	require.NoError(t, err, "Failed to upload large file for multipart upload test")
+
+	cases := []struct {
+		Name              string
+		PresignSeparately bool
+		Copy              bool
+	}{
+		{"presign all parts", false, false},
+		{"presign each part separately", true, false},
+		{"presign each part separately, copy", true, true},
```

Comment:
populate the struct fields - it will make it readable and not break if someone will add another field between Copy and Presign...

Reply:
ITYM "name the struct fields", so I did that.